app/scripts/background.js

```diff
@@ -731,18 +731,23 @@ function emitDappViewedMetricEvent(origin) {
   );
   const numberOfTotalAccounts = Object.keys(preferencesState.identities).length;
 
-  controller.metaMetricsController.trackEvent({
-    event: MetaMetricsEventName.DappViewed,
-    category: MetaMetricsEventCategory.InpageProvider,
-    referrer: {
-      url: origin,
+  controller.metaMetricsController.trackEvent(
```

Comment:
This comes from 12.16.2 hotfix https://github.com/MetaMask/metamask-extension/pull/32304